shared-ui/src/main/java/org/hzontal/shared_ui/bottomsheet/BottomSheetUtils.kt

```diff
@@ -440,44 +440,86 @@ BottomSheetUtils {
+                override fun onBind(holder: GenericSheetHolder) {
+                    with(holder) {
+                        title.text = titleText
+                        description.text = descriptionText
+                        actionButtonLabel?.let {
+                            actionButton.text = it
+                        }
+                        cancelButtonLabel?.let {
+                            cancelButton.text = it
+                        }
 
-                    actionButton.setOnClickListener {
-                        consumer.accept(isConfirmed = true)
-                        customSheetFragment.dismiss()
-                    }
+                        actionButton.setOnClickListener {
+                            consumer.accept(isConfirmed = true)
+                            customSheetFragment.dismiss()
+                        }
+
+                        cancelButton.setOnClickListener {
+                            consumer.accept(isConfirmed = false)
+                            customSheetFragment.dismiss()
+                        }
+
+                        actionButton.visibility =
+                            if (actionButtonLabel.isNullOrEmpty()) View.GONE else View.VISIBLE
 
-                    cancelButton.setOnClickListener {
-                        consumer.accept(isConfirmed = false)
-                        customSheetFragment.dismiss()
                     }
+                }
+            })
+
+        customSheetFragment.transparentBackground()
+        customSheetFragment.launch()
+    }
+
+
+    fun showProgressImportSheet(
+        fragmentManager: FragmentManager,
+        titleText: String?,
+        totalProgress: Int,
+        progressNumberText: String,
+        progressStatus: MutableLiveData<Int>,
+        cancelText: String,
+        lifecycleOwner: LifecycleOwner,
+        onCancelImport: (() -> Unit)
 
-                    actionButton.visibility =
-                        if (actionButtonLabel.isNullOrEmpty()) View.GONE else View.VISIBLE
+    ) {
+        val customSheetFragment =
+            CustomBottomSheetFragment.with(fragmentManager).page(R.layout.layout_progess_sheet)
+                .cancellable(true).statusBarColor(R.color.space_cadet)
+        customSheetFragment.holder(DownloadStatustHolder(),
+            object : CustomBottomSheetFragment.Binder<DownloadStatustHolder> {
+                @RequiresApi(Build.VERSION_CODES.N)
+                override fun onBind(holder: DownloadStatustHolder) {
+                    with(holder) {
+                        progressStatus.observe(lifecycleOwner) { status ->
+                            subtitle.text = "$status/$totalProgress $progressNumberText"
+                            val statusPercent = status * 100 / totalProgress
+                            circularProgress.setProgress(statusPercent, true)
+                            linearProgress.setProgress(statusPercent, true)
```

Comment:
This requires a min level of 24, consider adding the else case as our minimum is 19, maybe create an extension function?